src/hpc/traverse.c

```diff
@@ -92,20 +85,16 @@ static void BeginTraversal(TraversalState * traversal)
     traversal->listSize = 0;
     traversal->listCapacity = 10;
     traversal->listCurrent = 0;
-    traversal->previousTraversal = currentTraversal();
-    TLS(traversalState) = traversal;
 }
 
-static void EndTraversal(void)
+static void EndTraversal(TraversalState * traversal)
```

Comment:
Is there any reason to keep EndTraversal around?

Reply:
Hmm, not really...

Reply:
Was mainly curious if there were later plans for it. Is there any obvious "transversally finished" assertion checks that could be put in there? Doesn't matter if not.

Reply:
Can't really think of any right now, but perhaps @rbehrends can?

Actaully, In a way, the motivation for this PR *was* to get rid od EndTraversal: because if there was any erro during a traversal, we'd abort the traversal but never called EndTraversal, leading to a leak at best, and possibly worse.